src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/Extensions.cs

```diff
@@ -51,25 +51,47 @@ public static string GetTypeNavInfoNameOrEmpty(this ITypeSymbol typeSymbol)
 
         public static string GetProjectDisplayName(this Project project)
         {
-            if (project.Solution.Workspace is VisualStudioWorkspaceImpl workspace)
+            // If the project name is unambiguous within the solution, use that name. Otherwise, use the unique name
+            // provided by IVsSolution3.GetUniqueUINameOfProject. This covers all cases except for a single solution
+            // with two or more multi-targeted projects with the same name and same targets.
+            //
+            // https://github.com/dotnet/roslyn/pull/43800
+            // http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/949113
+
+            if (IsUnambiguousProjectNameInSolution(project))
             {
-                var hierarchy = workspace.GetHierarchy(project.Id);
-                if (hierarchy != null)
+                return project.Name;
+            }
+            else if (project.Solution.Workspace is VisualStudioWorkspaceImpl workspace
```

Comment:
```suggestion
            else if (project.Solution.Workspace is VisualStudioWorkspace workspace
```

You don't need to couple to the internal type, so no reason to do so.

Reply:
This was carried over from the old implementation, but now fixed.